.github/workflows/builder_go_slsa3.yml

```diff
@@ -329,7 +329,7 @@ jobs:
       contents: write # Needed to write artifacts to a release.
     runs-on: ubuntu-latest
     needs: [build-dry, build, provenance, detect-env]
-    if: inputs.upload-assets == true
+    if: startsWith(github.ref, 'refs/tags/') && inputs.upload-assets
```

Comment:
is this correct? I think we want users to be able to upload assets on non-tag triggers, like workflow_dispatch, so long as they can a tag-name to indicate which tag they us to upload to?
This allows some teams to create a release draft out-of-band and give us the tag.

Reply:
Ah, nice catch - agreed with the above, either tag or non-empty tag name. then the individual steps below switch on either case

Reply:
Yes, you're right I missed this. Updated.